`NetworkConfig` & `NodeConfig`

dahlia opened the pull request:
*Addresses #1946.*

The key purpose of those two classes is to let developers create a `Swarm<T>` instance with minimum settings, e.g.:

```csharp
var nodeConf = new NodeConfiguration<NullAction>();
await nodeConf.GetSwarm().StartAsync();
```

Both classes provide their default constructor, and these parameterless constructors configure themselves with sane defaults.

greymistcube commented:
Arg... there are some name clashes and differently imagined workflow. I was thinking more of a clear cut process, but now that `Libplanet.Node` has already been incorporated into `main`, it's probably best to scrap [libplanet-unity] branch. 😕

[libplanet-unity]: https://github.com/planetarium/libplanet/tree/libplanet-unity

greymistcube commented:
As we have discussed, can you make this target [libplanet-node] branch? Also to silo this project as much as possible from affecting the main branch, it'd be nice to have a separate PR for all changes to non-`Libplanet.Node` related stuff. 🙂

Let's keep this 2000th PR! 😛

[libplanet-node]: https://github.com/planetarium/libplanet/tree/libplanet-node

dahlia commented:
@greymistcube I changed its target branch to *libplanet-node*, and extracted unrelated changes to a [separated pull request](https://github.com/planetarium/libplanet/pull/2010)!
